src/components/ContentNode/LinkableHeading.vue

```diff
@@ -95,4 +95,8 @@ $icon-margin: 7px;
     }
   }
 }
+
+.visuallyhidden {
```

Comment:
@marinaaisa as written, I'm only doing this for the linkable heading scenario, but do you think it might make sense to lift this rule to _all_ visually hidden elements? I wasn't sure, so I just did it this way for now, but I wouldn't mind making the change for everything if that makes sense.

Reply:
Good question 🤔 I doubt those elements need to be selected.

Reply:
I agree that adding `user-select: none;` to `visually-hidden` would be useful.
But there is other problem: this CSS property only solves the case you only copy text of the title, if you start copying from some other text and continue with the title, the "in page link" is still there. 🤔

Reply:
One solution that comes to my mind is adding an event listener to `copy` or `select` that adds `display: none` (this prevents from copying but also prevents screen readers to read the content) to class `visually-hidden`. Then we will need to delete the `display` property when we consider that the action is done with another event listener (not sure in which)

Reply:
I think the "in page link" is still there because although the `span` is visually hidden and can't be selected, it's still part of the name of the link. When we copy the heading by selecting it, it's copying the entire link, not just the text (if you paste into Slack, it will format the result into a link automatically instead of plain text). 

This is what I see when I click & drag on the text portion of the link
<img width="265" alt="image" src="https://github.com/apple/swift-docc-render/assets/87735557/4be92f50-584f-45c4-bd42-82550995dc7e">

Reply:
Looks like we were using aria-labels for accessibility for the link, but changed it [here](https://github.com/apple/swift-docc-render/pull/698). I don't completely understand what was the issue with the aria-label in the original implementation? Is it just preventing scrolling to the correct position?

Reply:
Ah, I see what I'm missing now—this still happens on Chrome when using paste-with-formatting—not just a regular paste (without formatting).

Reply:
We could use this mechanism, its commonly used for line-numbers in code listings, which is how we use it too afaik.

```css
.link::before {
  content: attr(data-accessible-text);
}

Reply:
> Looks like we were using aria-labels for accessibility for the link, but changed it here. I don't completely understand what was the issue with the aria-label in the original implementation? Is it just preventing scrolling to the correct position?

@hqhhuang with the aria-label we were replacing the content of the headers for screen readers, so unfortunately it's not an option :/

Reply:
> We could use this mechanism, its commonly used for line-numbers in code listings, which is how we use it too afaik.
> 
> ```css
> .link::before {
>   content: attr(data-accessible-text);
> }
> ```

We do this on the `StrikeThrough.vue` too, it could be a solution. But it won't fix the `visuallyhidden` class issue in other cases of the web. We use this class to provide AX information in other places across the web and I guess we have the same issue with copy and pasting.

Reply:
Another alternative is using `display: none` instead of `visuallyhidden`. So it won't appear for copying, screen readers or visual users but we can reference it with an `id` that we can use in `aria-labelledby` after the id of the router-link itself, to contain the content of it too. I tried it and it works.

```vue
<router-link
  v-if="shouldLink"
  :to="{ hash: `#${anchor}` }"
  id="router-link"
  :aria-labelledby="`router-link ax-in-page-link-${_uid}`"
  class="header-anchor"
  @click="handleFocusAndScroll(anchor)"
>
  <slot />
  <span hidden :id="`ax-in-page-link-${_uid}`" >{{ $t('accessibility.in-page-link') }}</span>
  <LinkIcon class="icon" aria-hidden="true" />
</router-link>
```

It would be even better if we apply
```vue
 <span hidden id="ax-in-page-link" >{{ $t('accessibility.in-page-link') }}</span>
 ```
only once in the document, so we don't repeat that HTML node every time and we just make a reference to it.
Note: I used `_uid` in the one that it's inside the component because ids need to be unique.

Reply:
Thanks for all the help and ideas everyone!

I decided to go with the pseudo element approach for now, since that is what I was most familiar with from working on the strikethrough implementation before, but I'd be open to exploring the other approach in the future and seeing if it makes sense to refactor other usages of "visuallyhidden" with similar copy/paste issues.

Reply:
> > Looks like we were using aria-labels for accessibility for the link, but changed it here. I don't completely understand what was the issue with the aria-label in the original implementation? Is it just preventing scrolling to the correct position?
> 
> @hqhhuang with the aria-label we were replacing the content of the headers for screen readers, so unfortunately it's not an option :/

Could we resolve this by including the content of the header as part of the aria-label? Like setting "In page link: Overview" as a whole to be the aria label?

Reply:
> Could we resolve this by including the content of the header as part of the aria-label? Like setting "In page link: Overview" as a whole to be the aria label?

@hqhhuang The issue is that we use a `slot` for the content of the header so it's not a prop that we can attach in the aria-label. That's why I wrote a solution before using `aria-labelledby` instead, because it allow us to use the ids to reference the content inside the slot.